### PR TITLE
[#5167] Handle password-protected partner sites

### LINKED ISSUE #5167: Bug: Password protected partner sites
### What were you doing?

Users are not prompted to enter the password for the password-protected partner sites.

### What should've happened?

The password page should be displayed when the user is trying to access password-protected partner sites.

### My environment

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by zuhdil**
# TODO / Done

Summarize what has been changed / what has to be done in order to finalize the PR.

 - [x] Endpoint to show the Django-lockdown password page
 - [x] Handle password protection for project-directory (landing) page
 - [x] Handle password protection for public project page

# Test plan

What tests are necessary to ensure this works or doesn't break anything working

 - [ ] Test password-protected project directory page
 - [ ] Test password-protected public project page
 
---
Closes #5167

### CODE REVIEW

**Comment by zuhdil** [comment]
To @ifirmawan 

The endpoint: `/en/lockpass/?next=<url_to_redirect_when_password_correct>`
`next` param defaults to the `project-directory` page if None

frontend implementation is more or less like https://github.com/akvo/akvo-rsr/blob/production/akvo/rsr/spa/app/root.jsx#L49

File: akvo/rsr/dir/app/modules/index/view.jsx [hunk 1]
```diff
@@ -83,7 +83,10 @@ const View = () => {
         setFilters(defaults)
       }
     }
-  }, [apiData])
+    if (apiError && apiError.response && apiError.response.status === 403) {
+      window.location.href = '/en/lockpass/?next='
+    }
```
Comment (MichaelAkvo) [inline akvo/rsr/dir/app/modules/index/view.jsx:88]: I think this should only happen if the primary domain is `akvoapp.org`

File: akvo/rsr/dir/app/modules/project-page/ProjectPage.jsx [hunk 2]
```diff
@@ -84,6 +84,9 @@ const ProjectPage = ({ match: { params }, location }) => {
     }
     if ((loading && (apiError || projectError)) || (loading && user && !apiError)) {
       setLoading(false)
+      if (projectError && projectError.response && projectError.response.status === 403) {
+        window.location.href = `/en/lockpass/?next=${window.location.href}`
+      }
```
Comment (MichaelAkvo) [inline akvo/rsr/dir/app/modules/project-page/ProjectPage.jsx:89]: I think this should only happen if the primary domain is `akvoapp.org`

File: akvo/rsr/views/__init__.py [hunk 3]
```diff
@@ -14,3 +14,10 @@ def index(request):
     """Redirect user to project directory or My RSR."""
 
     return HttpResponseRedirect(reverse('project-directory', args=[]))
+
+
+def lockpass(request):
```
Comment (MichaelAkvo) [inline akvo/rsr/views/__init__.py:19]: I think a little documentation here would help future developers understand what's going on.

From reading this, one wouldn't know why it exists and what makes it show the password-protected page.

**Review by MichaelAkvo — CHANGES_REQUESTED** [review]

File: akvo/rsr/dir/app/utils/misc.js [hunk 4]
```diff
@@ -99,3 +100,5 @@ export const getLogo = logo => logo
 export const getFirstPhoto = photos => (photos && photos.length)
   ? photos.slice(0, 1).pop()
   : null
+
+export const isPartnerSites = () => (window.location.href.indexOf('localakvoapp') !== -1 || window.location.href.indexOf('akvoapp') !== -1)
```
Comment (MichaelAkvo) [inline akvo/rsr/dir/app/utils/misc.js:104]: You can use `location.hostname` for this: https://developer.mozilla.org/en-US/docs/Web/API/Location/hostname

e.g 

```js
window.location.hostname.endsWith("akvoapp.com") || window.location.hostname === "localakvoapp"
```

And it should be `isPartnerSite`

**Review by MichaelAkvo — CHANGES_REQUESTED** [review]
@ifirmawan one last small change

**Pull request merged**